### ISSUE TITLE
Way to clear old ACRA crash reports on device?

### ISSUE THREAD

**Issue by re3lex** [ACRA/acra#351]
Hi,
My app has ACRA enabled and recently I noticed that size of app data on my testing phone is about 100MB. I have found that 90% of that size is used by old ACRA crash reports (sure they were sent to my back end properly).
So I'm looking for a way to configure ACRA to drop sent reports.
Is there any way to do that?
Or what you could suggest?

**Comment by william-ferguson-au** [comment]
ACRA removes sent reports.

Can you post the list of files that are being retained by your app?

**Comment by re3lex** [comment]
@william-ferguson-au I see lots of files like 
1453589775015-IS_SILENT.stacktrace

I'm sending silent logs in some cases and during development I expect to get many crash reports but I also expect that sent logs should be removed after sending

**Comment by william-ferguson-au** [comment]
They are removed after sending.
Need to figure out why yours aren't.
Post you ACRA config.

**Comment by re3lex** [comment]
@william-ferguson-au Thanks for quick reply!
My ACRA is configured using annotation 

```
@ReportsCrashes(
  formUri = BuildConfig.CRASH_REPORT_URL,
  httpMethod = org.acra.sender.HttpSender.Method.POST,
  mode = ReportingInteractionMode.TOAST,
  resToastText = R.string.crash_toast_text)
```

**Comment by re3lex** [comment]
THat's good that automatic log removal feature is implemented and working in common case.
Let me rephrase this issue then: is there any way to invoke force log removal?
Saying I implement periodic task to invoke some ACRA API which will clear all logs?

**Comment by william-ferguson-au** [comment]
OK, ACRA is rate limited, so that it only sends at most 5 crash reports at a time. This was done in order to stop aberrant apps from buring through lots of bandwidth, flooding report servers and locking up the device because they went into infinite crash loop or just dumped inordinate numbers of silent reports.

But once a report has been sent it is removed.

If you are generating lots of silent reports, then  it is possible that they are accumulating.
ACRA was not designed to be generating high report loads. If you need that capacity then you should really look at an Analytics provider.

**Comment by re3lex** [comment]
@william-ferguson-au thanks for your explanation. It is useful for me.
I will revisit my app and reduce number of silent reports.

But I'm still looking for the way to clear old logs manually.
I investigated ACRA API but didn't find anything helpful for that.

**Comment by william-ferguson-au** [comment]
There is no way.
The solution is to use ACRA as it is designed.

You can manually clear the data for the app. THis will also remove any ACRA reports.

**Comment by re3lex** [comment]
ok, thanks
